### PR TITLE
[Multimodal] MMEFT: Add model specs for mmeft model

### PR DESCRIPTION
**Pull request by vivek-dani**

### CODE REVIEW

File: assets/models/system/mmeft/description.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ Task|Dataset|Python sample (Notebook)|CLI with YAML 
+ |--|--|--|--| 
+ Multimodal multi-class classification|TODO: add dataset|<a href='https://aka.ms/azureml-ft-sdk-multimodal-mc-classification' target='_blank'>multimodal-multiclass-classification.ipynb</a>|<a href='https://aka.ms/azureml-ft-cli-multimodal-mc-classification' target='_blank'>multimodal-multiclass-classification.sh</a> 
+ Multimodal multi-label classification |TODO: add dataset|<a href='https://aka.ms/azureml-ft-sdk-multimodal-ml-classification' target='_blank'>multimodal-multilabel-classification.ipynb</a>|<a href='https://aka.ms/azureml-ft-cli-multimodal-ml-classification' target='_blank'>multimodal-multilabel-classification.sh</a> 
+ 
+ ### Sample inputs and outputs (for real-time inference) 
+ #### Sample input 
+ ```json 
+ { 
+ 'input_data': {
```
Comment (rjaincc) [inline assets/models/system/mmeft/description.md:25]: Let's use double quotes for consistency.

File: assets/models/system/mmeft/spec.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+$schema: https://azuremlschemas.azureedge.net/latest/model.schema.json
+name: mmeft
```
Comment (rjaincc) [inline assets/models/system/mmeft/spec.yaml:2]:
```suggestion
name: MMEFT
```

File: assets/models/system/mmeft/description.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+     { 
+         'label1': 0.1,
+		 'label2': 0.7,
+		 'label3': 0.2
+     } 
+ ] 
+  
+ ``` 
+  
+ #### Model inference - visualization for a sample image
```
Comment (rjaincc) [inline assets/models/system/mmeft/description.md:46]:
```suggestion
 #### Model inference - visualization for a sample input
```

File: assets/models/system/mmeft/description.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+Multimodal Early Fusion Transformer (MMEFT): A Transformer based Architecture for Reasoning over Structured and Unstructured Data
```
Comment (rjaincc) [inline assets/models/system/mmeft/description.md:1]: I would recommend the following rephrasing:

Multi-Modal Early Fusion Transformer, MMEFT, is a transformer-based framework tailored for processing both structured and unstructured data. It's suitable for multi-class and multi-label multimodal classification tasks and is capable of handling datasets with features from diverse modes, including categorical, numerical, image, and text.

In MMEFT, the embedding Layer produces distinct non-contextual embeddings for different feature modes. Then, the fusion Layer integrates the non-contextual embeddings to yield contextual multimodal embeddings. The aggregation Layer consolidates the multimodal embeddings into a singular representation. Lastly, the output Layer, processes the final multimodal embedding to generate the model's prediction. MMEFT uses, utilizes BertTokenizer for text data embeddings, and mploys the 'openai/clip-vit-base-patch32' model from Hugging Face for image data embeddings.

This model is designed to offer a comprehensive approach to multimodal data, ensuring accurate and efficient classification across varied datasets.

NOTE: We highly recommend to finetune the model on your dataset before deploying.

File: assets/models/system/mmeft/spec.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+path: ./
+properties:
+  inference-min-sku-spec: 2|0|7|14
+  inference-recommended-sku: Standard_DS3_v2
+  model_id: mmeft
+tags:
+  Preview: ''
+  license: msft-mit
```
Comment (rjaincc) [inline assets/models/system/mmeft/spec.yaml:10]: why msft-mit?

File: assets/models/system/mmeft/spec.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+$schema: https://azuremlschemas.azureedge.net/latest/model.schema.json
+name: mmeft
+path: ./
+properties:
+  inference-min-sku-spec: 2|0|7|14
+  inference-recommended-sku: Standard_DS3_v2
```
Comment (rjaincc) [inline assets/models/system/mmeft/spec.yaml:6]: Should be list of skus:   
     Standard_DS3_v2, Standard_D4a_v4, Standard_D4as_v4, Standard_DS4_v2,
    Standard_D8a_v4, Standard_D8as_v4, Standard_DS5_v2, Standard_D16a_v4, Standard_D16as_v4,
    Standard_D32a_v4, Standard_D32as_v4, Standard_D48a_v4, Standard_D48as_v4, Standard_D64a_v4,
    Standard_D64as_v4, Standard_D96a_v4, Standard_D96as_v4, Standard_FX4mds, Standard_F8s_v2,
    Standard_FX12mds, Standard_F16s_v2, Standard_F32s_v2, Standard_F48s_v2, Standard_F64s_v2,
    Standard_F72s_v2, Standard_FX24mds, Standard_FX36mds, Standard_FX48mds, Standard_E2s_v3,
    Standard_E4s_v3, Standard_E8s_v3, Standard_E16s_v3, Standard_E32s_v3, Standard_E48s_v3,
    Standard_E64s_v3, Standard_NC4as_T4_v3, Standard_NC6s_v3, Standard_NC8as_T4_v3,
    Standard_NC12s_v3, Standard_NC16as_T4_v3, Standard_NC24s_v3, Standard_NC64as_T4_v3,
    Standard_NC24ads_A100_v4, Standard_NC48ads_A100_v4, Standard_NC96ads_A100_v4,
    Standard_ND96asr_v4, Standard_ND96amsr_A100_v4, Standard_ND40rs_v2

File: assets/models/system/mmeft/description.md (hunk 3 above)
Comment (vivek-dani) [reply assets/models/system/mmeft/description.md]: removed this section, We will add this section once visualizations are tested

File: assets/models/system/mmeft/spec.yaml (hunk 5 above)
Comment (vivek-dani) [reply assets/models/system/mmeft/spec.yaml:10]: In the link that you had shared about licenses, Microsoft version of MIT is little different from raw MIT license

Comment (rjaincc) [reply assets/models/system/mmeft/spec.yaml:10]: That's still MIT license, right? 
I would recommend keeping MIT as license - same as how we do it for HF and MMD models.

File: assets/models/system/mmeft/description.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  
+ ```json 
+ [ 
+     {
+		 "label1": 0.1,
+		 "label2": 0.7,
+		 "label3": 0.2
+     } 
+ ] 
+
```
Comment (rjaincc) [inline assets/models/system/mmeft/description.md:45]: nit: much out indentated?

**Review by rjaincc — DISMISSED** [review]
Approving with a minor comment.

File: assets/models/system/mmeft/description.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+ Multimodal multi-class classification|Airbnb listings dataset|<a href='https://aka.ms/azureml-ft-sdk-multimodal-mc-classification' target='_blank'>multimodal-multiclass-classification.ipynb</a>|<a href='https://aka.ms/azureml-ft-cli-multimodal-mc-classification' target='_blank'>multimodal-multiclass-classification.sh</a> 
+ Multimodal multi-label classification |Hateful memes dataset|<a href='https://aka.ms/azureml-ft-sdk-multimodal-ml-classification' target='_blank'>multimodal-multilabel-classification.ipynb</a>|<a href='https://aka.ms/azureml-ft-cli-multimodal-ml-classification' target='_blank'>multimodal-multilabel-classification.sh</a> 
+ 
+ ### Sample inputs and outputs (for real-time inference) 
+ #### Sample input 
+ ```json 
+ { 
+ "input_data": { 
+         "columns": ["column1","column2","column3","column4","column5","column6"], 
+         "data": [22,11.2,"It was a great experience!",<base 64 encoded image string>,"Categorical value",True]
```
Comment (PhaniShekhar) [inline assets/models/system/mmeft/description.md:29]: In other models, we are using something like "image1" in the "data" and adding a note at the bottom that image1 is a base64 encoded image string or publicly accessible url. Please check clip or other models.

Comment (PhaniShekhar) [inline assets/models/system/mmeft/description.md:29]: Should we add example with multiple (say 2) rows ? Is it supported ?

File: assets/models/system/mmeft/description.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+     } 
+ } 
+ ``` 
+  
+ #### Sample output 
+  
+  
+ ```json 
+ [
```
Comment (PhaniShekhar) [inline assets/models/system/mmeft/description.md:38]: Is this output inline with the new image classification mlflow model output ?

File: assets/models/system/mmeft/description.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
+Multimodal Early Fusion Transformer, MMEFT, is a transformer-based model tailored for processing both structured and unstructured data.
+
+It can be used for multi-class and multi-label multimodal classification tasks, and is capable of handling datasets with features from diverse modes, including categorical, numerical, image, and text.
+The MMEFT architecture is composed of embedding, fusion, aggregation, and output layers. The embedding layer produces independent non-contextual embeddings for features of varying modes. Then, the fusion Layer integrates the non-contextual embeddings to yield contextual multimodal embeddings. The aggregation layer consolidates these contextual multimodal embeddings into a single multimodal embedding vector. Lastly, the output Layer, processes the final multimodal embedding to generate the model's prediction based on task for which it is used. 
+MMEFT uses, utilizes BertTokenizer for text data embeddings, and considers 'openai/clip-vit-base-patch32' model from Hugging Face for image data embeddings.
```
Comment (PhaniShekhar) [inline assets/models/system/mmeft/description.md:5]: Nit: uses, utilizes ? Is it intentional ?

File: assets/models/system/mmeft/description.md [hunk 11]
```diff
@@ -0,0 +1,47 @@
+Multimodal Early Fusion Transformer, MMEFT, is a transformer-based model tailored for processing both structured and unstructured data.
```
Comment (PhaniShekhar) [inline assets/models/system/mmeft/description.md:1]: Could you please share a link to the model in any dev registry ?

File: assets/models/system/mmeft/description.md (hunk 8 above)
Comment (PhaniShekhar) [inline assets/models/system/mmeft/description.md:29]: Does the non-finetuned model work out of the box with tabular data ? Which tabular featurizer does it use ?

**Review by PhaniShekhar — CHANGES_REQUESTED** [review]

File: assets/models/system/mmeft/spec.yaml [hunk 12]
```diff
@@ -0,0 +1,12 @@
+$schema: https://azuremlschemas.azureedge.net/latest/model.schema.json
+name: MMEFT
```
Comment (PhaniShekhar) [inline assets/models/system/mmeft/spec.yaml:2]: Just for my understanding, Is this the name we are using in the model catalog ?

File: assets/models/system/mmeft/spec.yaml [hunk 13]
```diff
@@ -0,0 +1,17 @@
+$schema: https://azuremlschemas.azureedge.net/latest/model.schema.json
+name: mmeft
+path: ./
+properties:
+  evaluation-min-sku-spec: 4|1|28|176,
```
Comment (rjaincc) [inline assets/models/system/mmeft/spec.yaml:5]: no commas

**Review by PhaniShekhar — APPROVED** [review]

**Pull request merged**